src/Vault.sol

```diff
@@ -0,0 +1,101 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity ^0.8.24;
+
+import {BaseVault} from "src/BaseVault.sol";
+import {FeeMath} from "src/module/FeeMath.sol";
+import {IStrategy} from "src/interface/IStrategy.sol";
+import {Math} from "./Common.sol";
+
+contract Vault is BaseVault {
+    using Math for uint256;
+
+    error ExceedsMaxBasisPoints();
+
+    bytes32 public constant FEE_MANAGER_ROLE = keccak256("FEE_MANAGER_ROLE");
+
+    struct FeeStorage {
+        /// @notice The base withdrawal fee in basis points (1e8 = 100%)
+        uint64 baseWithdrawalFee;
+    }
+    /// @notice The fraction of the buffer below which flat fees apply, in basis points (1e8 = 100%). Only used by quadratic fees
+
+    function _getFeeStorage() internal pure returns (FeeStorage storage $) {
+        assembly {
+            $.slot := 0xde924653ae91bd33356774e603163bd5862c93462f31acccae5f965be6e6599b
+        }
+    }
+
+    /**
+     * @notice Initializes the vault.
+     * @param admin The address of the admin.
+     * @param name The name of the vault.
+     * @param symbol The symbol of the vault.
+     */
+    function initialize(
+        address admin,
+        string memory name,
+        string memory symbol,
+        uint8 decimals_,
+        uint64 baseWithdrawalFee_,
+        bool countNativeAsset_
+    ) external virtual initializer {
+        __ERC20_init(name, symbol);
+        __AccessControl_init();
+        __ReentrancyGuard_init();
+        _grantRole(DEFAULT_ADMIN_ROLE, admin);
+
+        VaultStorage storage vaultStorage = _getVaultStorage();
+        vaultStorage.paused = true;
+        vaultStorage.decimals = decimals_;
```

Comment:
i got to be missing the asset initialization

Reply:
you're right we separated it into a setter we should always bootstrap with 1 asset